classes/sample_set.php

```diff
@@ -66,6 +78,16 @@ public function add_sample(\ExcimerLogEntry $sample) {
             $this->samples[] = $sample;
             $this->counter = 0;
         }
+        // Each time a sample is added, recalculate the maxstackdepth for this set.
+        foreach ($this->samples as $sample) {
+            $trace = $sample->getTrace();
+            if ($trace) {
+                $stackdepth = count($trace);
+                if ($stackdepth > $this->maxstackdepth) {
+                    $this->maxstackdepth = $stackdepth;
+                }
+            }
+        }
```

Comment:
I believe you don't need to look through all the samples again to find the maximum. You'll only need to look at the current maximum and the current sample (to be added).

I also think instead of doing
```
                if ($stackdepth > $this->maxstackdepth) {
```
You can probably get by with something like `$this->maxstackdepth = max($this->maxstackdepth, count($trace)`; (keeps nesting to a minimum)